### ISSUE TITLE
Tor2web doesn't work when only in http

### ISSUE THREAD

**Issue by fpietrosanti** [globaleaks/Tor2web#250]
We found ourself in using Tor2web only in HTTP, with TRANSLATION and dummy proxy but the software was not working properly giving the following error:
root@tor2web:/etc# /etc/init.d/tor2web start
 * Starting Tor2web tor2web...                                                                                                                               * Starting tor daemon...                                                                                                                            [ OK ]
                                                                                                                                                     [ OK ]
root@tor2web:/etc# Unhandled error in Deferred:
Unhandled Error
Traceback (most recent call last):
Failure: twisted.spread.pb.PBConnectionLost: [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
]
Unhandled error in Deferred:
Unhandled Error
Traceback (most recent call last):
Failure: twisted.spread.pb.PBConnectionLost: [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
]
Unhandled error in Deferred:
Unhandled Error
Traceback (most recent call last):
Failure: twisted.spread.pb.PBConnectionLost: [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
]
Unhandled error in Deferred:
Unhandled Error
Traceback (most recent call last):
Failure: twisted.spread.pb.PBConnectionLost: [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
]
Unhandled error in Deferred:
Unhandled Error
Traceback (most recent call last):
Failure: twisted.spread.pb.PBConnectionLost: [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
]
Unhandled error in Deferred:
Unhandled Error
Traceback (most recent call last):
root@tor2web:/etc# logout
Connection to 10.119.0.100 closed.

**Comment by virgil** [comment]
The most recent OnionLink patch fixes this.

**Comment by virgil** [comment]
move to close.

**Comment by NSkelsey** [comment]
Commit 22bf385 fixes.

**Comment by virgil** [comment]
I'm getting this error again on a fresh test installation on Ubuntu 14.04.

`Linux version 4.4.0-42-generic (buildd@lcy01-01) (gcc version 4.8.4 (Ubuntu 4.8.4-2ubuntu1~14.04.3) ) #62~14.04.1-Ubuntu SMP Fri Oct 7 23:15:48 UTC 2016`

Still figuring out how to fix.

**Comment by evilaliv3** [comment]
Thanks Virgil, nice catch.

I will iterate in a slightly different patch because i just noticed other small errors related to this.

**Comment by evilaliv3** [comment]
Issue solved; 

Fix shipped in release 3.1.54

**Comment by vjrj** [comment]
I still have this issue with 3.1.54 in trusty:

```
Unhandled error in Deferred:
Unhandled Error
Traceback (most recent call last):
Failure: twisted.spread.pb.PBConnectionLost: [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
]
```

**Comment by evilaliv3** [comment]
@vjrj is it possible to give a look at the not private details of your tor2web.conf?

**Comment by vjrj** [comment]
Here we go:
```
[main]

## TOR2WEB CONFIGURATION FOR GLOBALEAKS
# This is a minimal configuration template for a Tor2web to be used
# with globaleaks installed on the same server.
# WARNING: That setup is only valid for mid/low-risks deployments, such as anticorruption compliance in western countries

# Hostname of the public website
basehost = xxx.yyy.zzz

# Local IP address Tor2web must be listening
listen_ipv4 = aa.bb.cc.dd

# Tor Hidden Service 
# To get it issue the following command
# cat /var/globaleaks/torhs/hostname 
mode = TRANSLATION
onion = xxyyzz.onion

# This allows Tor2web to work as a proxy for a local GlobaLeaks platform exposed on http://127.0.0.1:8082
dummyproxy = http://127.0.0.1:8082

# CERTIFICATE CONFIGURATION
# For the commands needed for generating the files below specified refer to the wiki page at:
# https://github.com/globaleaks/Tor2web/wiki/Installation-Guide
ssl_key = /home/tor2web/certs/tor2web-key.pem
#
# LOAD THE .CRT FILE THAT HAS BEEN GIVEN YOU BY YOUR CA
ssl_cert = /home/tor2web/certs/tor2web-cer.pem
#
# LOAD HERE INTERMEDIATE CERTIFICATE CHAIN OF YOUR CA
# PLEASE BE CAREFUL TO PUT THE CERT CHAIN IN THE RIGHT ORDER!!
ssl_intermediate = /home/tor2web/certs/tor2web-intermediate.pem

# IF GLOBALEAKS IS USED IN EMBEDDED MODE UNCOMMENT THIS
# AND READ https://github.com/globaleaks/GlobaLeaks/wiki/Integration-Guide
# disable_tor_redirection = True
# blockhotlinking = False

# IF YOU WANT TO SERVE TORBROWSER FILES FROM THIS SERVER
# COMMENT THIS OUT and READ https://github.com/globaleaks/Tor2web/wiki/GetTor
disable_gettor = True

# If set to True will disable the tor2web disclaimer
disable_disclaimer = True

# If set to True will disable the tor2web banner
disable_banner = True

```

**Comment by evilaliv3** [comment]
@vjrj so you are running configured for using HTTPS.

this ticket is instead dealing with the configuration without HTTPS but only HTTP.

may i guess that SSL certificates have not been created? can you look that all the path that you have put in the configuration point to an existing file readable by the tor2web user?

stupidly the conde right now does not perform any valiudation on them at the bootup, in order to recognize the main errors, but i'm planning to implement that in order to guide the user.

**Comment by vjrj** [comment]
You are right. 

Now it's working after change:
`ssl_cert = /home/tor2web/certs/tor2web-cer.pem`
by:
`ssl_cert = /home/tor2web/certs/tor2web-certificate.pem`
in my config. My fault. 

By the way, if you want to improve the template/doc, I followed this wiki page:
https://github.com/globaleaks/GlobaLeaks/wiki/Installing-GlobaLeaks-Tor2web-Letsencrypt
but `/usr/share/tor2web/data/conf/tor2web-globaleaks.conf` has `tor2web-cer.pem` instead of `tor2web-certificate.pem`.

Thanks indeed

**Comment by fpietrosanti** [comment]
If i remember correctly to fix this it would be useful to fix #155 and #262 that require Tor2web to "validate" that certificate's related files configured exists, are accessible and are valid before starting

**Comment by caztillo** [comment]
thank you  @virgil for your solution. Indeed the wiki needs to change ssl_cert = `/home/tor2web/certs/tor2web-cer.pem`
to 
`ssl_cert = /home/tor2web/certs/tor2web-certificate.pem`
